.github/actions/setup/action.yml

```diff
@@ -0,0 +1,40 @@
+#
+# Github Action to setup the libcgroup directory
+#
+# Copyright (c) 2021 Oracle and/or its affiliates.
+# Author: Tom Hromatka <tom.hromatka@oracle.com>
+#
+
+#
+# This library is free software; you can redistribute it and/or modify it
+# under the terms of version 2.1 of the GNU Lesser General Public License as
+# published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+# for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, see <http://www.gnu.org/licenses>.
+#
+
+name: Setup the libseccomp directory
+description: "Install dependencies and configure libseccomp"
+runs:
+  using: "composite"
+  steps:
+  - run: sudo apt-get install build-essential valgrind clang-tools lcov gperf astyle codespell
+    shell: bash
+  - run: |
+      sudo apt-get install -y python3.6 python3-distutils python3-setuptools python3-pip
+      python3 -m pip install --upgrade pip
+      python3 -m pip install cython
+      # Add cython to the path
+      echo "/home/runner/.local/bin" >> $GITHUB_PATH
+    shell: bash
+  - run: |
+      ./autogen.sh
+      ./configure --enable-python
+      make check-build
```

Comment:
I don't think there is any doubt that running `./autogen.sh` is the right thing to do here, but I'm less convinced about the `configure` and `make` commands; should we relocate those to the individual CI jobs, even if it results in some duplication among separate jobs?

Reply:
I completely agree.  For the initial version, I tried to keep the Github Actions logic the same as the Travis logic.

But since the overall design is so much different (runs are split up by test type), your suggestion is definitely a better solution.